modules/engine/lib/engine/http/request.js

```
@@ -298,6 +297,7 @@ function sendHttpRequest(client, options, args, start, timings, reqStart, key, c
             }
         });
         res.on('end', function () {
+            happy = true;
```

Added as suggested. Also, throwing a warn event instead of console log.